lib/mapper.js

```diff
@@ -15,39 +38,92 @@ var objCase = require('obj-case');
  * @api private
  */
 
-exports.track = function(track){
-  var props = track.properties();
-  if (track.revenue()) {
-    props.amount = track.revenue();
-    objCase.del(props, 'revenue');
-  }
-
+exports.track = function(track) {
   return {
     email: track.email(),
     userId: track.userId(),
     eventName: track.event(),
     createdAt: unixTime(track.timestamp()),
-    dataFields: props
+    dataFields: track.properties()
   };
 };
 
 /**
- * Map `identify`.
+ * Map `Added Product and Removed Product`.
  *
- * @param {Identify} track
+ * This is only invoked when a full products array is present
+ *
+ * @param {Track} track
  * @return {Object}
  * @api private
  */
 
-exports.identify = function(identify){
-  var traits = traverse(clone(identify.traits()));
-  var created = identify.created();
-  var userId = identify.userId();
-  objCase.del(traits, 'id')
-  if (created) traits.met = created.toISOString();
+exports.addedProduct =
+exports.removedProduct = function(track) {
+  var props = track.properties({ revenue: 'amount' });
   return {
-    email: identify.email(),
-    userId: userId,
-    dataFields: traits
+    user: {
+      email: track.email(),
+      userId: track.userId()
+    },
+    items: formatProducts(track.products())
+  };
+}
+
+/**
+ * Map `Completed Order`.
+ *
+ * @param {Track} track
+ * @return {Object}
+ * @api private
+ */
+
+exports.completedOrder = function(track) {
+  return {
+    createdAt: unixTime(track.timestamp()),
+    total: track.total(),
+    user: {
+      userId: track.userId(),
+      email: track.email()
+    },
+    dataFields: foldl(function(acc, val, key) {
+      if (key !== 'total' && key !== 'products') acc[key] = val;
+      return acc;
+    }, {}, track.properties({ revenue: 'amount' })),
```

Comment:
Ah, I see you replied above that it returns with ```revenue``` aliased to ```amount```. Can we take that out? I meant to remove ```sanitizeProps``` in my original PR but forgot to do it. Thanks/sorry!